doc/source/serve/deploy_model_tutorial.rst

```diff
@@ -0,0 +1,329 @@
+====================================
+End-to-End Model Deployment Tutorial
+====================================
+
+By the end of this tutorial you will have learned how to deploy a machine
+learning model locally via Ray Serve.
+
+First, install Ray Serve and all of its dependencies by running the following
+command in your Terminal:
+
+.. code-block:: bash
+
+  $ pip install "ray[serve]"
+
+For this tutorial, we'll use `HuggingFace's SummarizationPipeline <https://huggingface.co/docs/transformers/main_classes/pipelines#transformers.SummarizationPipeline>`_
+to access a model that summarizes text. Let's take a look at how it works.
+This is the code for the model:
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_local.py
+  :language: python
+  :start-after: __local_model_start__
+  :end-before: __local_model_end__
+
+The Python file, called ``local_model.py`` uses the ``summarize`` function to
+generate summaries of text. The ``summarizer`` variable inside ``summarize``
+points to a function that uses the
+`t5-small <https://huggingface.co/t5-small>`_ model to summarize text.
+When ``summarizer`` is called on a Python String, it returns summarized text
+encapsulated within a dictionary and a list. ``summarize`` then removes the
+dictionary and list, and it returns only the summary. The file can be run
+locally by executing the Python script, which uses the model to summarize an
+article about the Apollo 11 moon landing.
+
+.. code-block:: bash
+
+  $ python local_model.py
+
+  "two astronauts steered their fragile lunar module safely and smoothly to the
+  historic landing . the first men to reach the moon -- Armstrong and his
+  co-pilot, col. Edwin E. Aldrin Jr. of the air force -- brought their ship to
+  rest on a level, rock-strewn plain ."
+
+This tutorial's goal is to deploy this model using Ray Serve, so it can be
+queried over HTTP. We'll start by converting the above Python function into a
```

Comment:
```suggestion
This tutorial's goal is to deploy this model using Ray Serve, so it can be scaled up and
queried over HTTP. We'll start by converting the above Python function into a
```